pkg/controller/atlasdeployment/advanced_deployment.go

```diff
@@ -222,3 +224,29 @@ func GetAllDeploymentNames(client mongodbatlas.Client, projectID string) ([]stri
 	}
 	return deploymentNames, nil
 }
+
+func normalizeInstanceSize(currentInstanceSize string, autoscaling *mdbv1.AdvancedAutoScalingSpec) string {
+	currentSize := extractNumberFromInstanceTypeName(currentInstanceSize)
+	minSize := extractNumberFromInstanceTypeName(autoscaling.Compute.MinInstanceSize)
+	maxSize := extractNumberFromInstanceTypeName(autoscaling.Compute.MaxInstanceSize)
+
+	if currentSize < minSize {
+		return autoscaling.Compute.MinInstanceSize
+	}
+
+	if currentSize > maxSize {
+		return autoscaling.Compute.MaxInstanceSize
+	}
+
+	return currentInstanceSize
+}
+
+func extractNumberFromInstanceTypeName(instanceTypeName string) int {
+	name := strings.TrimPrefix(instanceTypeName, "M")
+	name = strings.TrimPrefix(name, "R")
+	name = strings.TrimSuffix(name, "_NVME")
+
+	number, _ := strconv.Atoi(name)
```

Comment:
Why ignore an error here? We do not have any kubebuilder validations on the `instanceSizeName` field. I'd say it's better to explicitly return an error and put it into the status that the instance size name is incorrect.

Reply:
This code probably won't handle `M50` and `M50_NVME` instances